lib/src/async_message_grouper.dart

```diff
@@ -56,7 +56,7 @@ class AsyncMessageGrouper implements MessageGrouper {
   }
 
   /// Stop listening to the stream for further updates.
-  Future cancel() {
+  Future? cancel() {
```

Comment:
Why is this nullable?

Reply:
Because StreamQueue.cancel returns nullable future: https://dart.googlesource.com/async/+/refs/tags/2.5.0-nullsafety.3/lib/src/stream_queue.dart#402. But looking at the stream queue sources looks like it should not return null: _cancel returns null only when called for the second time (https://dart.googlesource.com/async/+/refs/tags/2.5.0-nullsafety.3/lib/src/stream_queue.dart#503), so added a `!`